protocol-units/bridge/contracts/src/AtomicBridgeInitator.sol

```diff
@@ -83,8 +97,19 @@ contract AtomicBridgeInitiator is IAtomicBridgeInitiator, Initializable {
         if (bridgeTransfer.state != MessageState.INITIALIZED) revert BridgeTransferStateNotInitialized();
         if (block.number < bridgeTransfer.timeLock) revert TimeLockNotExpired();
         bridgeTransfer.state = MessageState.REFUNDED;
+        // Decrease balance of originator and transfer WETH back to originator
+        balances[bridgeTransfer.originator] -= bridgeTransfer.amount;
         if (!weth.transfer(bridgeTransfer.originator, bridgeTransfer.amount)) revert WETHTransferFailed();
 
         emit BridgeTransferRefunded(bridgeTransferId);
     }
+
+    // Counterparty contract to withdraw WETH for originator
+    function withdrawWETH(address originator, uint256 amount) external {
+        require(msg.sender == counterpartyContract, "Caller is not the counterparty contract");
+        require(balances[originator] >= amount, "Insufficient balance");
+        balances[originator] -= amount;
```

Comment:
The balances should exactly match, unless a recipient has multiple in flight. I would just treat it as a pool.

Reply:
I've made the check exactly `==`.

What data structure do you propose for the pool? I've seen mappings as a pretty common data structure for storing liquidity in a contract.

Reply:
See comments above, I would just pool.